### PR TITLE
Feature/realsense trigger

### PR DESCRIPTION
**Pull request by michaelpantic**

### CODE REVIEW

File: platforms/nuttx/nuttx-configs/px4fmu-v2/nsh/defconfig [hunk 1]
```diff
@@ -571,7 +571,8 @@ CONFIG_STM32_USART_SINGLEWIRE=y
 # SPI Configuration
 #
 # CONFIG_STM32_SPI_INTERRUPTS is not set
-# CONFIG_STM32_SPI_DMA is not set
+CONFIG_STM32_SPI_DMA=y
+CONFIG_STM32_SPI4_DMA=y
```
Comment (ZacharyTaylor) [inline platforms/nuttx/nuttx-configs/px4fmu-v2/nsh/defconfig]: The newer PX4 boards use px4fmu-v3, the only difference I am aware of is it has more onboard memory, so hopefully we can make the same changes there.

File: src/drivers/imu/adis16448/adis16448.cpp [hunk 2]
```diff
@@ -224,6 +226,7 @@ class ADIS16448 : public device::SPI
 
 	uint16_t			_product;	/** product code */
 
+	work_s			_work{};
```
Comment (ZacharyTaylor) [inline src/drivers/imu/adis16448/adis16448.cpp]: why the empty initialization list?

File: src/drivers/imu/adis16448/adis16448.cpp [hunk 3]
```diff
@@ -560,6 +567,16 @@ ADIS16448::ADIS16448(int bus, const char *path_accel, const char *path_gyro, con
 	_mag_scale.z_scale  = 1.0f;
 
 	memset(&_call, 0, sizeof(_call));
+
+	// allocate dma memory
+	_dma_data_buffer = (struct ADISReport *)fat_dma_alloc(13*sizeof(uint16_t));
```
Comment (ZacharyTaylor) [inline src/drivers/imu/adis16448/adis16448.cpp]: maybe a named constexpr for the 13?

File: src/drivers/imu/adis16448/adis16448.cpp [hunk 4]
```diff
@@ -560,6 +567,16 @@ ADIS16448::ADIS16448(int bus, const char *path_accel, const char *path_gyro, con
...
+
+	if(!_dma_data_buffer){
+		PX4_ERR("malloc failed");
+	}
+	else{
+		PX4_ERR("malloc succeded");
```
Comment (ZacharyTaylor) [inline src/drivers/imu/adis16448/adis16448.cpp]: Probably don't throw an error on success

File: platforms/nuttx/nuttx-configs/px4fmu-v2/nsh/defconfig (hunk 1 above)
Comment (michaelpantic) [reply platforms/nuttx/nuttx-configs/px4fmu-v2/nsh/defconfig]: In the internal build structure of px4, the px4fmu-v3 architecture uses the px4fmu-v2 definitions - so these changes automatically apply to anything on the px3fmu-v3 build chain too.

File: src/drivers/imu/adis16448/adis16448.cpp (hunk 3 above)
Comment (michaelpantic) [reply src/drivers/imu/adis16448/adis16448.cpp]: that is still horrible hacked-in - so will need to change this anyway  
This is still sort of proof-of-concept code from the very beginning - I wasn't sure if the dma_alloc etc would work at all and postponed cleaning up this to sort of now.. :-)

File: src/drivers/imu/adis16448/adis16448.cpp (hunk 4 above)
Comment (michaelpantic) [reply src/drivers/imu/adis16448/adis16448.cpp]: same as above

File: src/drivers/imu/adis16448/adis16448.cpp (hunk 2 above)
Comment (michaelpantic) [reply src/drivers/imu/adis16448/adis16448.cpp]: hacky way of initializing everything to zero - seen in other parts of nuttx/px4 drivers that use the work queue

**Comment by michaelpantic** [comment]
@ZacharyTaylor  FYI It's cleaned up now and tested again on hardware, if you want to take a second look. Otherwise I merge later today.

File: src/drivers/imu/adis16448/adis16448.cpp [hunk 5]
```diff
@@ -559,7 +566,17 @@ ADIS16448::ADIS16448(int bus, const char *path_accel, const char *path_gyro, con
...
 
-	memset(&_call, 0, sizeof(_call));
+	// allocate dma memory
+	#if defined(CONFIG_FAT_DMAMEMORY)
+		_dma_data_buffer = (struct ADISReport *)fat_dma_alloc(sizeof(ADISReport));
+	#else
+		_dma_data_buffer = nullptr;
+	#endif
+
+	if(_dma_data_buffer == nullptr){
```
Comment (ZacharyTaylor) [inline src/drivers/imu/adis16448/adis16448.cpp]: If this happens will the code still run correctly or would it be better to just have the driver exit?

Comment (michaelpantic) [reply src/drivers/imu/adis16448/adis16448.cpp]: yep should just assign normal memory if dma module is not built or dma alloc failed

**Pull request merged**